pkg/controller/volume/persistentvolume/index.go

```diff
@@ -157,6 +170,26 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *v1.PersistentVol
 	return nil, nil
 }
 
+// checkVolumeModeMatches is a convenience method that checks volumeMode for PersistentVolume
+// and PersistentVolumeClaims along with making sure that the Alpha feature gate BlockVolume is
+// enabled.
+// This is Alpha and could change in the future.
+func checkVolumeModeMatches(pvcSpec v1.PersistentVolumeClaimSpec, pvSpec v1.PersistentVolumeSpec) (bool, bool) {
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		requestedVolumeMode := pvcSpec.VolumeMode
+		pvVolumeMode := pvSpec.VolumeMode
+
+		if requestedVolumeMode != pvVolumeMode {
```

Comment:
You should compare the contents of the pointers instead of the pointer values?

Reply:
yeah, I fixed that